### PR TITLE
feat: add m68k parser and queries

### PR DESCRIPTION
**Pull request by grahambates**
Adding my parser for Motorola 68000 assembly.

### CODE REVIEW

File: queries/m68k/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  (address_register)
+  (data_register)
+  (float_register)
+  (named_register)
+] @keyword
+
+(repeat (control_mnemonic) @repeat)
+(conditional (control_mnemonic) @conditional)
+
+(interpolated (macro_arg) @embedded)
```
Comment (theHamsta) [inline queries/m68k/highlights.scm:50]: there is no `@embedded`. You probably want to use `@none`

**Review by theHamsta — COMMENTED** [review]
Seems to look ok, apart from that little comment

File: queries/m68k/highlights.scm (hunk 1 above)
Comment (grahambates) [reply queries/m68k/highlights.scm]: I've removed it. The previous query mapping `macro_arg` to `variable.builtin` is fine in this context too.

Comment (theHamsta) [reply queries/m68k/highlights.scm]: You probably meant to use `injections.scm`. See how injections work for C. It is not in `highlight.scm` for our repo

Comment (grahambates) [reply queries/m68k/highlights.scm]: Thanks for that. I had a look and I don't think it's really suitable here as it's a different use case to C's macro args. I'm happy to explain in more detail, but in short I think it makes sense to leave all macro args highlighted as variables. I did however add injections for comments and I've now added locals queries too, which I've tried out with `nvim-treesitter-refactor`.

**Pull request merged**